api/api/accounts/serializers.py

```diff
@@ -49,3 +49,84 @@ def create(self, validated_data):
             Staff.objects.create(user=user, **staff_data)
 
         return user
+
+    def update(self, instance, validated_data):
+        student_data = validated_data.pop('student', None)
+        faculty_data = validated_data.pop('faculty', None)
+        staff_data = validated_data.pop('staff', None)
+
+        self._update_base_user(instance, validated_data)
+
+        try:
+            student = instance.student
+        except Exception:
+            student = None
+
+        try:
+            faculty = instance.faculty
+        except Exception:
+            faculty = None
+
+        try:
+            staff = instance.staff
+        except Exception:
+            staff = None
+
+        if student:
+            self._update_student(student, student_data)
+        elif student_data:
+            Student.objects.create(user=instance, **student_data)
+        elif faculty:
+            self._update_faculty(faculty, faculty_data)
+        elif faculty_data:
+            Faculty.objects.create(user=instance, **faculty_data)
+        elif staff:
+            self._update_staff(staff, staff_data)
+        elif staff_data:
+            Staff.objects.create(user=instance, **staff_data)
+
+        return instance
+
+    def _update_base_user(self, instance, validated_data):
+        instance.email = validated_data.get('email', instance.email)
+        instance.first_name = validated_data.get('first_name', instance.first_name)
+        instance.last_name = validated_data.get('last_name', instance.last_name)
+        instance.gender = validated_data.get('gender', instance.gender)
+        instance.contact_no = validated_data.get('contact_no', instance.contact_no)
+        instance.user_type = validated_data.get('user_type', instance.user_type)
+        instance.is_active = validated_data.get('is_active', instance.is_active)
+        instance.picture_url = validated_data.get('picture_url', instance.picture_url)
```

Comment:
Can we do something like:

```python
updatable_fields = ['email', 'first_name', 'last_name', ...]

for field in updatable_fields:
  field_stored_value = getattr(instance, field)
  field_new_value = validated_data.get(field, field_stored_value)
  setattr(instance, field, field_new_value)
```

Maybe extract this piece in a helper method:
```python
def _update_instance(self, instance, validated_data, updatable_fields):
  ...
```

Reply:
This is much better.